### PR TITLE
[BugFix] Entity Recreation + Parenting

### LINKED ISSUE #600: [Bug] Entity recreation causes Parenting systems to fail.
How to replicate:
Create an entity.
Delete the entity a few frames later.
Re-create the same entity at the same index.

Known affected Systems:
-UiParentSystem #591 [Fixed]
-TransformSystem [Pending #601]

Impact:
As those systems think that it is the same entity, when it is not, they do not execute changes that should be applied to those newly created entities.

If you know of a System using a Bitset to track entities by their index, it is most likely affected by this too.

### PR DESCRIPTION
**Pull request by jojolepro**
Fixes #600

---
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/601)

### CODE REVIEW

**Comment by Aceeri** [comment]
I don't think frame init needs to be changed. You can't have 2 entities of the same generation in the same loop.

File: amethyst_core/src/transform/systems.rs [hunk 1]
```diff
@@ -70,11 +71,18 @@ impl<'a> System<'a> for TransformSystem {
 
         {
             // Checks for entities with a modified local transform or a modified parent, but isn't initialized yet.
-            let filter = locals.open().0 & parents.open().0 & !&self.init; // has a local, parent, and isn't initialized.
-            for (entity, _) in (&*entities, &filter).join() {
+            let filter = locals.open().0 & parents.open().0; // has a local, parent, and isn't initialized.
+            let mut to_add = Vec::<Entity>::new();
+            for (entity, _) in (&*entities, &filter)
+                .join()
+                .filter(|&(e, _)| !self.init.contains(&e))
```
Comment (torkleyy) [inline amethyst_core/src/transform/systems.rs]: This is pretty bad for performance :(

Comment (jojolepro) [reply amethyst_core/src/transform/systems.rs]: O(n) RIP :/

**Comment by jojolepro** [comment]
Which data structure would be best for this one? HashSet?

**Comment by torkleyy** [comment]
The bitshet should stay and get updated by FlaggedStorage.

**Comment by jojolepro** [comment]
Will be fixed when the new specs releases. This will also include the performance fix for the ui parenting system. See the ui layout PR for more info. #591

**Comment by jojolepro** [comment]
Being worked on by something else to fix multiple bugs at once.

**Pull request closed without merging**